alica_engine/src/engine/AlicaTimer.cpp

```diff
@@ -0,0 +1,68 @@
+#include "engine/AlicaTimer.h"
+
+namespace alica
+{
+
+class AlicaSystemTimer::AlicaSystemTimerImpl
+{
+public:
+    using TimerCb = std::function<void()>;
+
+    AlicaSystemTimerImpl(TimerCb&& userCb, alica::AlicaTime period)
+            : _userCb(std::move(userCb))
+            , _period(period.inMilliseconds())
+            , _isActive(false)
+    {
+    }
+
+    ~AlicaSystemTimerImpl()
+    {
+        if (_isActive) {
+            stop();
+        }
+    }
+
+    void start()
+    {
+        if (_isActive) {
+            stop();
+        }
+        _isActive = true;
+        _thread = std::thread([this]() {
```

Comment:
I think we should put a comment indicating this timer uses 1 thread per timer (i.e. 1 thread per behaviour/plan) which is far from ideal & should be eventually replaced with a thread pool implementation if it needs to be really useful

Reply:
1 thread could lead to it being overloaded if there are many high frequency behaviours. Why is a thread pool bad though? You can also set the thread pool size to any number you want

Reply:
I think a single thread would be enough to run the clock, but could have more threads for the callbacks themselves.  Anyway for another time

Reply:
Sleeping thread shouldn't be a performance problem IMHO. Even a dozen of it.